upgrade.go

```diff
@@ -525,8 +574,8 @@ func (pth *programTestHelper) previewOnlyUpgradeTest(stateFile string) (finalErr
 		require.True(t, opts.SkipUpdate, "expecting SkipUpdate: true")
 		require.True(t, opts.SkipRefresh, "expecting SkipRefresh: true")
 		require.True(t, opts.SkipExportImport, "expecting SkipExportImport: true")
-		require.Falsef(t, opts.SkipEmptyPreviewUpdate,
-			"expecting SkipEmptyPreviewUpdate: false")
+		require.Truef(t, opts.SkipEmptyPreviewUpdate,
```

Comment:
Side-effect of this change is that it now uses only one call to pulumi preview instead of two. It used to do two calls for bad reasons (unable to call into ProgramTest the way we want here), without tweaking ProgramTest.